### PR TITLE
default to sauvola-ms-split and win-size=DPI

### PR DESCRIPTION
**Pull request by bertsky**

### CODE REVIEW

**Comment by bertsky** [comment]
Of course, the test images under `assets/data/scribo-cli` need to be updated again. The difference is barely perceptible on the sauvola images run in the CI test, but this has a large (and positive) impact on Niblack and Singh, and generally on non-300-DPI images or heterogeneous layouts.

@kba could you please update the images again (for all algorithms)?

File: ocrd-olena-binarize [hunk 1]
```diff
@@ -79,13 +76,43 @@ function image_id_from_fpath {
...
     if ! image_in_id=$(xmlstarlet "${options[@]}" | head -1); then
-        warning "image URL '${image_in_fpath}' not referenced"\
-                "in METS for input file ID=${in_id} (pageId=${in_pageId})"
+        ocrd log warning "image URL '${image_in_fpath}' not referenced in METS for input file ID=${in_id} (pageId=${in_pageId})" >&2
         image_in_id="${in_id}-IMG" # fallback
     fi
     echo "$image_in_id"
 }   
 
+function auto_winsize {
```
Comment (kba) [inline ocrd-olena-binarize:85]: clever.

**Review by kba — APPROVED** [review]

**Comment by kba** [comment]
>  @kba could you please update the images again (for all algorithms)?

sure

File: ocrd-olena-binarize (hunk 1 above)
Comment (bertsky) [reply ocrd-olena-binarize:85]: Hah! I have completely forgotten about this corner. This was neither anticipation of https://github.com/OCR-D/spec/pull/164 not recycling – I swear!

**Comment by kba** [comment]
CI check fails because `sudo make install PREFIX=/usr/local` will skip actually installing `ocrd-olena-binarize` script. The last command is `cp ocrd-tool.json /usr/local/share/ocrd_olena`. I'm at my wit's end here for today, maybe someone can spot the error.

**Comment by mikegerber** [comment]
> CI check fails because `sudo make install PREFIX=/usr/local` will skip actually installing `ocrd-olena-binarize` script. The last command is `cp ocrd-tool.json /usr/local/share/ocrd_olena`. I'm at my wit's end here for today, maybe someone can spot the error.

That's (probably) the problem I encountered before: I think `TOOLS` is empty because this:

~~~
TOOLS = $(shell ocrd ocrd-tool ocrd-tool.json list-tools 2>/dev/null)
~~~

fails silently. Then `install` does nothing here in this dependency:

~~~
install: $(TOOLS:%=$(BINDIR)/%)
~~~

**Comment by mikegerber** [comment]
In master (haven't tested it with this PR), I get this:

~~~
[root@b-pc30533 ocrd_olena]# make install PREFIX=/usr/local
test -x /usr/local/bin/scribo-cli && \
/usr/local/bin/scribo-cli sauvola --help >/dev/null 2>&1 || \
	make build-olena
which ocrd >/dev/null 2>&1 || \
	/usr/bin/pip3 install ocrd # needed for ocrd CLI (and bashlib)
WARNING: Running pip install with root privileges is generally not a good idea. Try `pip3 install --user` instead.
Collecting ocrd
  Using cached https://files.pythonhosted.org/packages/47/e0/4f17f17aa9dbe7c11d571f2f364d04ef987d075368fe0595088012b129eb/ocrd-2.12.6-py3-none-any.whl
Requirement already satisfied: ocrd-validators==2.12.6 in /usr/local/lib/python3.8/site-packages (from ocrd) (2.12.6)
Requirement already satisfied: pyyaml in /usr/lib64/python3.8/site-packages (from ocrd) (5.3.1)
Requirement already satisfied: Deprecated==1.2.0 in /usr/local/lib/python3.8/site-packages (from ocrd) (1.2.0)
Requirement already satisfied: atomicwrites>=1.3.0 in /usr/local/lib/python3.8/site-packages (from ocrd) (1.4.0)
Requirement already satisfied: ocrd-models==2.12.6 in /usr/local/lib/python3.8/site-packages (from ocrd) (2.12.6)
Requirement already satisfied: bagit-profile>=1.3.0 in /usr/local/lib/python3.8/site-packages (from ocrd) (1.3.1)
Requirement already satisfied: Flask in /usr/local/lib/python3.8/site-packages (from ocrd) (1.1.2)
Requirement already satisfied: ocrd-modelfactory==2.12.6 in /usr/local/lib/python3.8/site-packages (from ocrd) (2.12.6)
Requirement already satisfied: bagit>=1.7.0 in /usr/local/lib/python3.8/site-packages (from ocrd) (1.7.0)
Requirement already satisfied: click>=7 in /usr/local/lib/python3.8/site-packages (from ocrd) (7.1.2)
Requirement already satisfied: jsonschema in /usr/lib/python3.8/site-packages (from ocrd) (3.2.0)
Requirement already satisfied: ocrd-utils==2.12.6 in /usr/local/lib/python3.8/site-packages (from ocrd) (2.12.6)
Requirement already satisfied: lxml in /usr/lib64/python3.8/site-packages (from ocrd) (4.4.1)
Requirement already satisfied: requests in /usr/lib/python3.8/site-packages (from ocrd) (2.22.0)
Requirement already satisfied: opencv-python-headless in /usr/local/lib64/python3.8/site-packages (from ocrd) (4.3.0.36)
Requirement already satisfied: shapely in /usr/local/lib64/python3.8/site-packages (from ocrd-validators==2.12.6->ocrd) (1.7.0)
Requirement already satisfied: wrapt<2,>=1 in /usr/local/lib/python3.8/site-packages (from Deprecated==1.2.0->ocrd) (1.12.1)
Requirement already satisfied: itsdangerous>=0.24 in /usr/local/lib/python3.8/site-packages (from Flask->ocrd) (1.1.0)
Requirement already satisfied: Jinja2>=2.10.1 in /usr/lib/python3.8/site-packages (from Flask->ocrd) (2.11.2)
Requirement already satisfied: Werkzeug>=0.15 in /usr/local/lib/python3.8/site-packages (from Flask->ocrd) (1.0.1)
Requirement already satisfied: attrs>=17.4.0 in /usr/lib/python3.8/site-packages (from jsonschema->ocrd) (19.3.0)
Requirement already satisfied: pyrsistent>=0.14.0 in /usr/lib64/python3.8/site-packages (from jsonschema->ocrd) (0.16.0)
Requirement already satisfied: setuptools in /usr/lib/python3.8/site-packages (from jsonschema->ocrd) (41.6.0)
Requirement already satisfied: six>=1.11.0 in /usr/lib/python3.8/site-packages (from jsonschema->ocrd) (1.14.0)
Requirement already satisfied: numpy>=1.17.0 in /usr/lib64/python3.8/site-packages (from ocrd-utils==2.12.6->ocrd) (1.18.4)
Requirement already satisfied: Pillow>=7.2.0 in /usr/local/lib64/python3.8/site-packages (from ocrd-utils==2.12.6->ocrd) (7.2.0)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /usr/lib/python3.8/site-packages (from requests->ocrd) (3.0.4)
Requirement already satisfied: idna<2.9,>=2.5 in /usr/lib/python3.8/site-packages (from requests->ocrd) (2.8)
Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/lib/python3.8/site-packages (from requests->ocrd) (1.25.7)
Requirement already satisfied: MarkupSafe>=0.23 in /usr/lib64/python3.8/site-packages (from Jinja2>=2.10.1->Flask->ocrd) (1.1.1)
Installing collected packages: ocrd
Successfully installed ocrd-2.12.6
~~~
So /usr/local/bin/ocrd-olena-binarize is not installed.

It installs when running it a second time:
~~~
[root@b-pc30533 ocrd_olena]# make install PREFIX=/usr/local
test -x /usr/local/bin/scribo-cli && \
/usr/local/bin/scribo-cli sauvola --help >/dev/null 2>&1 || \
	make build-olena
which ocrd >/dev/null 2>&1 || \
	/usr/bin/pip3 install ocrd # needed for ocrd CLI (and bashlib)
sed 's|^SHAREDIR=.*|SHAREDIR="/usr/local/share/ocrd_olena"|;s|^PYTHON=.*|PYTHON="/usr/bin/python3"|' ocrd-olena-binarize > /usr/local/bin/ocrd-olena-binarize
chmod a+x /usr/local/bin/ocrd-olena-binarize
you already have '/usr/local/bin' in your PATH
~~~

One could say installing `ocrd` there in Makefile is to late because `TOOLS` is already empty. Not sure why this was not noticed earlier.

**Comment by mikegerber** [comment]
My Makefile kung fu is not good enough to tell if this is fixable.

**Comment by kba** [comment]
Thanks, that is helpful. I was suspecting the TOOLS magic, though elegant, was brittle. I'll replace it with a more explicit logic to install the executable.

**Pull request merged**